src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -313,5 +321,159 @@ public static bool TryParse(string value, out SemanticVersion result)
         {
             return !(b < a);
         }
+
+        // private methods
+        private bool IsServerVersion(string preRelease)
+        {
+            if (preRelease == null)
+            {
+                return false;
+            }
+
+            var pattern = @"^((?<releaseCandidate>rc\d+)?-?(?<internalBuild>\d+-g[0-9a-f]{4,40})?)$";
+            var match = Regex.Match(preRelease, pattern);
+
+            return match.Groups["releaseCandidate"].Success || match.Groups["internalBuild"].Success;
+        }
+
+        // nested types
+        internal class ServerVersion
```

Comment:
Rename class to `ServerInternalBuildVersion`

This class is ONLY suitable to represent server internal builds.

Reply:
I believe with the new approach it is not relevant.
Not done.